src/models/data.py

```diff
@@ -284,20 +369,39 @@ def __next__(self) -> Tuple[Tuple[Union[np.ndarray, torch.Tensor],
                     cur_max_idx = min(cur_max_idx + 1, self.max_idx)
 
                 out_x.append(arrays.x.historical)
+
+                out_x_curr.append(arrays.x.current)
                 out_x_add.append(arrays.x.pred_months)
+                out_x_add = [x_add for x_add in out_x_add if x_add is not None]
```

Comment:
What is the purpose of this?

Reply:
Can this filtering happen once, at the end? Or something like

```python
if arrays.x.current is not None:
    out_x_curr.append(arrays.x.current)
```

This prevents us from having to loop through the list of `out_x_add` and `out_x_curr` for every file we load.

Reply:
thanks!